prow/plugins/update_config/update_config.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package update_config
+
+import (
+	"fmt"
+
+	"github.com/Sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/kube"
+	"k8s.io/test-infra/prow/plugins"
+)
+
+const (
+	pluginName = "updater"
+	configFile = "prow/config.yaml"
+	pluginFile = "prow/plugins.yaml"
+)
+
+func init() {
+	plugins.RegisterPullRequestHandler(pluginName, handlePullRequest)
+}
+
+type githubClient interface {
+	CreateComment(owner, repo string, number int, comment string) error
+	GetPullRequestChanges(pr github.PullRequest) ([]github.PullRequestChange, error)
+	GetFile(org, repo, filepath string) (string, error)
+}
+
+type kubeClient interface {
+	ReplaceConfigMap(name string, config kube.ConfigMap) (kube.ConfigMap, error)
+}
+
+func handlePullRequest(pc plugins.PluginClient, pre github.PullRequestEvent) error {
+	return handle(pc.GitHubClient, pc.KubeClient, pc.Logger, pre)
+}
+
+func getRemoteFile(gc githubClient, org, repo, filepath string) (string, error) {
+	return gc.GetFile(org, repo, filepath)
+}
+
+func handleConfig(gc githubClient, kc kubeClient, org, repo string) error {
+	if content, err := getRemoteFile(gc, org, repo, configFile); err != nil {
+		return err
+	} else {
```

Comment:
Drop the extra indent, don't use an else block if you're returning error in the if.